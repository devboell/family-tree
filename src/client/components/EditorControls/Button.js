import styled from 'styled-components'


const Button = styled.button`
  padding: 3px 10px;
  margin-right: 20px;
  font-size: 12px;
  cursor: pointer;
`

export default Button
