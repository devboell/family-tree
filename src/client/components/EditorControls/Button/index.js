import styled from 'styled-components'


const Button = styled.button`
  padding: 3px 10px;
  margin-right: 20px;
  font-size: 12px;
`

export default Button
