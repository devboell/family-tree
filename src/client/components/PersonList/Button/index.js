import styled from 'styled-components'


const Wrapper = styled.button`
  width: 100%;
  background-color: rgb(76, 175, 80);
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;

  &:hover {
    background-color: gray;
  }
`

export default Wrapper
