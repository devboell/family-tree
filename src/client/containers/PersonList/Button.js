import styled, { css } from 'styled-components'

const hover = css`
  background-color: gray;
`

const hoverAndSelected = css`
  background-color: rgb(18, 106, 22);
  cursor: default;
`

const Button = styled.button`
  width: 100%;
  background-color: ${props => (
    props.isSelected
    ? 'rgb(18, 106, 22)'
    : 'rgb(76, 175, 80)'
  )};
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
  outline: 0;
  cursor: pointer;

  &:hover {
    ${props => (
      props.isSelected
      ? hoverAndSelected
      : hover
    )};
  }
`
Button.displayName = 'Button'

export default Button
