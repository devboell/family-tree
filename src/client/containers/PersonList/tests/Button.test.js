import 'jest-styled-components'
import React from 'react'

import Button from '../Button'


describe('Button component', () => {
  it('renders when selected', () => {
    expect(shallow(<Button isSelected />)).toMatchSnapshot()
  })

  it('renders when not selected', () => {
    expect(shallow(<Button isSelected={false} />)).toMatchSnapshot()
  })
})
