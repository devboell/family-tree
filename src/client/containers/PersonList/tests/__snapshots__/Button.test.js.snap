// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders when not selected 1`] = `
.cEzdAu {
  width: 100%;
  background-color: rgb(76, 175, 80);
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
  outline: 0;
  cursor: pointer;
}

.cEzdAu:hover {
  background-color: gray;
}

<button
  className="sc-bdVaJa cEzdAu"
/>
`;

exports[`Button component renders when selected 1`] = `
.ljGfAc {
  width: 100%;
  background-color: rgb(18, 106, 22);
  border: none;
  color: white;
  padding: 15px 32px;
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 12px;
  outline: 0;
  cursor: pointer;
}

.ljGfAc:hover {
  background-color: rgb(18, 106, 22);
  cursor: default;
}

<button
  className="sc-bdVaJa ljGfAc"
/>
`;
